hs_core/views/__init__.py

```diff
@@ -208,6 +211,15 @@ def update_metadata_element(request, shortkey, element_name, element_id, *args,
 
     return HttpResponseRedirect(request.META['HTTP_REFERER'])
 
+@api_view(['GET'])
+def file_download_url_mapper(request, shortkey, filename):
+    """ maps the file URIs in resourcemap document to django_irods download view function"""
+
+    authorize(request, shortkey, view=True, edit=True, full=True, superuser=True)
```

Comment:
authorize() is called but not processed here, so it appears to me that it does not really serve its purpose here. If authorize() needs to be called here, it needs to only map to download view when it is indeed authorized, otherwise, just raise an not-authorized exception or whatever makes sense for the specific workflow. Then perhaps in download() view, we can add in a parameter like check_permission with default value being true. Then in this case, since permission is already checked outside download function, a check_permission can be set to false being passed in via request.GET. I think we should not make duplicate calls unless there is no way around it, but understand in this specific case, the overhead of checking permission may not be much. However, after our access control is fully implemented, the overhead of checking permissions may increase, so I think we should eliminate duplicate calls when possible.